cli/lsp/language_server.rs

```diff
@@ -1319,23 +1320,135 @@ impl Inner {
     })
   }
 
+  fn refresh_workspace_files(&mut self) {
+    let enable_settings_hash = self.config.settings.enable_settings_hash();
+    if self.workspace_files_hash == enable_settings_hash {
+      return;
+    }
+    self.workspace_files.clear();
+    let document_preload_limit =
+      self.config.workspace_settings().document_preload_limit;
+    let mut pending = VecDeque::new();
+    let mut entry_count = 0;
+    if document_preload_limit > 0 {
```

Comment:
Overall, this PR looks good, but this code isn't unit testable anymore and we've lost those tests for this functionality. Could we make this unit testable again?

Reply:
Added back the unit test for the filtration in the function... caught a pretty bad bug in the process 😓 (missing negation at https://github.com/denoland/deno/pull/22937/commits/d5e821cf8f25eb4566dc82e66686aa5a4fb0af86#diff-20db93e8dfbff58c3024a4e247e639bf84fd5e8d2a447e8bbed9ee8942d688afR1336)